sample-extension-plugin/src/test/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunnerRestIT.java

```diff
@@ -85,6 +91,99 @@ public void testJobUpdateWithRescheduleJob() throws Exception {
     }
 
+    public void testJobUpdateWithRescheduleJobThenListJobs() throws Exception {
+
+        String SCHEDULER_INFO_URI = "/_plugins/_job_scheduler/api/jobs?by_node";
+
+        String index = createTestIndex();
+        SampleJobParameter jobParameter = new SampleJobParameter();
+        jobParameter.setJobName("sample-job-it");
+        jobParameter.setIndexToWatch(index);
+        jobParameter.setSchedule(new IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES));
+        jobParameter.setLockDurationSeconds(120L);
+
+        for (int i = 0; i < 10; i++) {
+            // Creates a new watcher job.
+            String indexN = createTestIndex();
+            SampleJobParameter jobParameterN = new SampleJobParameter();
+            jobParameterN.setJobName("sample-job-it" + i);
+            jobParameterN.setIndexToWatch(indexN);
+            jobParameterN.setSchedule(new IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES));
+            jobParameterN.setLockDurationSeconds(120L);
+
+            String jobIdN = OpenSearchRestTestCase.randomAlphaOfLength(10);
+            createWatcherJob(jobIdN, jobParameterN);
+        }
+
+        // Creates a new watcher job.
+        String jobId = OpenSearchRestTestCase.randomAlphaOfLength(10);
+        SampleJobParameter schedJobParameter = createWatcherJob(jobId, jobParameter);
+
+        // update the job params to now watch a new index.
+        String newIndex = createTestIndex();
+        jobParameter.setIndexToWatch(newIndex);
+
+        // wait till the job runner runs for the first time after 1 min & inserts a record into the watched index & then update the job with
+        // new params.
+        waitAndCreateWatcherJob(schedJobParameter.getIndexToWatch(), jobId, jobParameter);
+        long actualCount = waitAndCountRecords(newIndex, 130000);
+
+        // Asserts that the job runner has the updated params & it inserted the record in the new watched index.
+        Assert.assertEquals(1, actualCount);
+        long prevIndexActualCount = waitAndCountRecords(index, 0);
+
+        // Asserts that the job runner no longer updates the old index as the job params have been updated.
+        Assert.assertEquals(1, prevIndexActualCount);
+
+        Response response = makeRequest(client(), "GET", SCHEDULER_INFO_URI, Map.of(), null);
+        Map<String, Object> responseJson = JsonXContent.jsonXContent.createParser(
+            NamedXContentRegistry.EMPTY,
+            LoggingDeprecationHandler.INSTANCE,
+            response.getEntity().getContent()
+        ).map();
+
+        List<Map<String, Object>> nodes = (List<Map<String, Object>>) responseJson.get("nodes");
+        assertNotNull("Nodes list should not be null", nodes);
+        assertEquals(11, responseJson.get("total_jobs"));
+        assertEquals(0, ((List<?>) responseJson.get("failures")).size());
+        assertFalse("Should have at least one node", nodes.isEmpty());
+
+        for (Map<String, Object> node : nodes) {
```

Comment:
Let's limit the assertion to the total size and failures being 0. IMO I think the loop below is overkill for this test.